### PR TITLE
Replace our use of JS string `+` with template literal

### LINKED ISSUE #836: Use JavaScript/TypeScript template strings
From @jpwalbran on Slack:

> This is a PSA about using typescript template strings, which are things that we haven’t been using and totally should. They allow for having custom formatted strings with values attached that are cleaner than using
> ```
> myString = 'string literal' + value + 'another string literal';
> ```
> Instead, you can use a template string to turn that same expression into
> ```
> myString = `string literal ${value} another string literal`;
> ```
> More information on them can be found [here](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html). 

We totally should be using/modelling this here in the template and in the labs.

### PR DESCRIPTION
**Pull request by NicMcPhee**
I'm just using VS Code's global search tool to look for +s in JS/TS code, and then convert them to use template literals.

The Karma tests didn't fail if I broke many of these, but the E2E tests did, which was nice since I forgot to switch to backticks at least once and actually broke things. The tests caught it, though, and I think this is good now.

Closes #836

### CODE REVIEW

File: client/src/app/users/add-user.component.ts [hunk 1]
```diff
@@ -106,9 +106,11 @@ export class AddUserComponent implements OnInit {
 
   submitForm() {
     this.userService.addUser(this.addUserForm.value).subscribe(newID => {
-      this.snackBar.open('Added User ' + this.addUserForm.value.name, null, {
-        duration: 2000,
-      });
+      this.snackBar.open(
+        `Added User ${this.addUserForm.value.name}`,
```
Comment (helloworld12321) [inline client/src/app/users/add-user.component.ts:110]: Thoughts on lowercase here?

```suggestion
        `Added user ${this.addUserForm.value.name}`,
```

**Review by helloworld12321 — CHANGES_REQUESTED** [review]
I'd suggest one letter case change, but otherwise looks great, thanks! 🙂

**Review by helloworld12321 — APPROVED** [review]
LGTM! (Although you may need to re-run Cypress)

**Comment by NicMcPhee** [comment]
Poop – I'm pretty sure this actually breaks the test, which I'm guessing has that text (with the "wrong" case) hardwired in. Doh!

I'll fix that and equest a re-review.

**Review by helloworld12321 — APPROVED** [review]

**Pull request merged**